client/landing/stepper/declarative-flow/internals/steps-repository/free-post-setup/index.tsx

```diff
@@ -31,13 +31,16 @@ const FreePostSetup: Step = ( { navigation } ) => {
 	const [ selectedFile, setSelectedFile ] = useState< File | undefined >();
 	const [ isLoading, setIsLoading ] = useState( false );
 	const [ isSubmitError, setIsSubmitError ] = useState( false );
-
+	const [ siteDataFetched, setSiteDataFetched ] = useState( false );
 	const { saveSiteSettings } = useDispatch( SITE_STORE );
 
 	useEffect( () => {
-		setComponentSiteTitle( site?.name || '' );
-		setTagline( site?.description || '' );
-	}, [ site ] );
+		if ( site && ! siteDataFetched ) {
+			setComponentSiteTitle( site?.name || '' );
+			setTagline( site?.description || '' );
+			setSiteDataFetched( true );
+		}
+	}, [ site, siteDataFetched, setSiteDataFetched ] );
```

Comment:
I wonder if we'd be able to circumvent the `siteDataFetched` call by being more explicit with our `useEffect` dependencies. 🤔 

```
useEffect( () => { // Do Stuff }, [ site.name, site.description ] )
```

Reply:
Im still not completely sure whats causing this issue. The state is properly updated after the `site` data is retrieved, but the first key press is somehow ignored. I tried your suggestion `useEffect( () => { // Do Stuff }, [ site.name, site.description ] )` and also tried:

```
useEffect( () => {
 if ( site ) {
    setComponentSiteTitle( site?.name || '' );
    setTagline( site?.description || '' );
  }
}, [ site ] );
```
I need to look into the `setupForm` component a bit more to see if I find a clear explanation for this issue.